core/src/main/java/com/amplifyframework/predictions/models/TextFeature.java

```diff
@@ -28,20 +28,41 @@
  */
 @SuppressWarnings("unchecked")
 public abstract class TextFeature<T> extends Feature<T> {
-    private final TargetText target;
+    private final String targetText;
+    private final int startIndex;
+    private final int length;
 
     TextFeature(Builder<?, ? extends TextFeature<T>, T> builder) {
         super(builder);
-        this.target = builder.getTarget();
+        this.targetText = builder.getTargetText();
+        this.startIndex = builder.getStartIndex();
+        this.length = targetText.length();
     }
 
     /**
-     * Gets the target text and associated index.
-     * @return the target text
+     * Gets the target text to which this feature applies.
+     * @return the target portion of the input text
      */
     @NonNull
-    public final TargetText getTarget() {
-        return target;
+    public final String getTargetText() {
+        return targetText;
+    }
+
+    /**
+     * Gets the starting position of the target text
+     * with respect to the full input text.
```

Comment:
So the `targetText` starts at `startIndex` within a (>= size) input text?

Reply:
yep!